client/export_test.go

```diff
@@ -26,19 +27,37 @@ var (
 )
 
 func (client *Client) SetDoer(d doer) {
-	client.doer = d
+	client.Requester().(*defaultRequester).doer = d
```

Comment:
Should this `SetDoer` method be moved to a test method/function? I believe external packages can't actually call it anyway, as the `doer` interface isn't exported. But happy to leave this to another PR if you prefer.

Reply:
I think this aligns well with the general refactor proposal for this test framework based on https://github.com/canonical/pebble/issues/315